source/jormungandr/tests/journey_common_tests.py

```diff
@@ -1208,3 +1207,108 @@ def test_strange_line_name(self):
+        assert len(response['journeys']) == 1
+        assert len(response['journeys'][0]['sections']) == 1
+        assert response['journeys'][0]['sections'][0]['mode'] == 'bike'
+        assert response['journeys'][0]['sections'][0]['duration'] == 62
+
+    def test_first_section_mode_bike_and_last_section_mode_walking(self):
+        query = '{sub_query}&last_section_mode[]=bike&first_section_mode[]=walking&' \
+                'datetime={datetime}'.format(sub_query=sub_query, datetime="20120614T080000")
+        response = self.query_region(query)
+        self.is_valid_journey_response(response, query)
+        assert len(response['journeys']) == 2
+        assert len(response['journeys'][0]['sections']) == 3
+        assert response['journeys'][0]['sections'][0]['mode'] == 'walking'
+        assert response['journeys'][0]['sections'][0]['duration'] == 17
+        assert response['journeys'][0]['sections'][2]['mode'] == 'bike'
+        assert response['journeys'][0]['sections'][2]['duration'] == 21
+
+        assert len(response['journeys'][1]['sections']) == 1
+        assert response['journeys'][1]['sections'][0]['mode'] == 'walking'
+        assert response['journeys'][1]['sections'][0]['duration'] == 276
+
+    def test_last_section_mode_bake_only(self):
+        query = '{sub_query}&last_section_mode[]=bike&datetime={datetime}'.format(sub_query=sub_query,
+                                                                                  datetime="20120614T080000")
+        response = self.query_region(query)
+        self.is_valid_journey_response(response, query)
+        assert len(response['journeys']) == 2
+        assert len(response['journeys'][0]['sections']) == 3
+        assert response['journeys'][0]['sections'][0]['mode'] == 'walking'
+        assert response['journeys'][0]['sections'][0]['duration'] == 17
+        assert response['journeys'][0]['sections'][2]['mode'] == 'bike'
+        assert response['journeys'][0]['sections'][2]['duration'] == 21
+
+        assert len(response['journeys'][1]['sections']) == 1
+        assert response['journeys'][1]['sections'][0]['mode'] == 'walking'
+        assert response['journeys'][1]['sections'][0]['duration'] == 276
+
+    def test_first_section_mode_walking_and_last_section_mode_bike(self):
+        query = '{sub_query}&last_section_mode[]=walking&first_section_mode[]=bike&' \
+                'datetime={datetime}'.format(sub_query=sub_query, datetime="20120614T080000")
+        response = self.query_region(query)
+        self.is_valid_journey_response(response, query)
+        assert len(response['journeys']) == 1
+        assert len(response['journeys'][0]['sections']) == 1
+        assert response['journeys'][0]['sections'][0]['mode'] == 'bike'
+        assert response['journeys'][0]['sections'][0]['duration'] == 62
+
+    def test_first_section_mode_bike_only(self):
+        query = '{sub_query}&first_section_mode[]=bike&datetime={datetime}'.format(sub_query=sub_query,
```

Comment:
you should also test with only `first=[bike, walking]` and there should be a filtering (the idea is to test mor the default value, I may miss some cases also)

Reply:
ok

Reply:
ok

Reply:
☑️ done